app/questionnaire/questionnaire_schema.py

```diff
@@ -94,11 +96,75 @@ def __init__(
         self._populate_answer_dependencies()
         self._populate_when_rules_section_dependencies()
         self._populate_calculated_summary_section_dependencies()
+        self._populate_min_max_for_numeric_answers()
 
     @cached_property
     def answer_dependencies(self) -> ImmutableDict[str, set[AnswerDependent]]:
         return ImmutableDict(self._answer_dependencies_map)
 
+    @cached_property
+    def min_and_max_map(self) -> ImmutableDict[str, str | dict[str, str]]:
+        return ImmutableDict(self._min_and_max_map)
+
+    def _set_min_max_value_for_answer(
+        self, value: str | dict[str, str], min_max: Any, longest_string: str | None
+    ) -> str:
+        if isinstance(value, int):
+            if longest_string and len(str(value)) > len(longest_string):
+                return str(value)
+            elif not longest_string:
+                return str(value)
+
+        elif isinstance(value, dict) and value:
+            if (
+                value.get("source") == "answers"
+                and value["identifier"] in self._min_and_max_map
+            ):
+                if longest_string and len(
+                    str(self._min_and_max_map[value["identifier"]][min_max])
+                ) > len(longest_string):
+                    return str(
+                        self._min_and_max_map[value["identifier"]][min_max]
+                    )
+                elif not longest_string:
+                    return str(
+                        self._min_and_max_map[value["identifier"]][min_max]
+                    )
```

Comment:
Any reason for this to return the string instead of the length of the string?. Doing so means you can simplify some of this logic.

Reply:
Its because Im also using it then to set the value on line 147